pkgs/applications/video/handbrake/default.nix

```diff
@@ -76,6 +77,13 @@ _EOF
 
     patchShebangs scripts
 
+    # On Darwin, use the Nix-provided libxml2 instead of the patched version
```

Comment:
these Darwin/Linux comments are confusing, neither line looks at the platform, so other platforms presumably get the Nix libxml2 as well?

Reply:
If you look at [the file that’s being patched](https://github.com/HandBrake/HandBrake/blob/master/libhb/module.defs#L48-L64) you see that the patch labeled as Darwin here only applies to Darwin, Cygwin, and MinGW, and since Nix only supports Darwin out of those three, the patch only applies to Darwin. Similarly, the patch labeled as being for Linux is only applied on systems that are not FreeBSD, NetBSD, Darwin, Cygwin, or MinGW, which I believe means that it will only be applied under Linux.

With that being said, though, I agree that this is not really clear just from reading the derivation file. Since it could get messed up in the future if the upstream changes how things work on different platforms, it would be best to change the derivation so that it only makes certain patches when building for certain OSes.

After the discussion in HandBrake/HandBrake#3048, though, I think we should be using the HandBrake-patched versions of libxml2. The HandBrake developers do make an effort to push their changes upstream, and while we can debate whether shipping vendored dependencies is a good idea, I think it’s definitely not a good idea to swap one of HandBrake’s dependencies for a different version and then expect HandBrake to continue to work correctly. When I have some time I’ll look into getting the HandBrake derivation to use the HandBrake-patched version of libxml2.

Reply:
Agreed, it's my fault, sorry. This was the first thing I did for nixpkgs and I split handbrake into the component packages because I thought it was cleaner. In reality many of those libraries are abandoned and applying the handbrake patches might not necessarily be a good idea.

Reply:
Since Handbrake does a lot of package patching.
And since we can not impose those patches on the Nixpkgs, but they are expected in Handbrake.

It seems the solution is for Handbrake to have an overlay/override set for its dependencies.

Reply:
not necessarily - it's been a while, but I believe it just keeps its own dependencies in its own tree and it builds them too. Adding an overlay seems like a lot of extra work.

Reply:
> it just keeps its own dependencies in its own tree
Yep.
> it builds them too
Nope, we hack it to and force to use Nixpkgs dependencies:
https://github.com/NixOS/nixpkgs/blob/35521e4ea7a5bb471ce6b7286f5ce951b13a1027/pkgs/applications/video/handbrake/default.nix#L82-L83
dynamic deps is a policy of Nixpkgs.

Yes, Handbrake tries to build its deps shop by default, but it is in our shop )

What to do further is the question.

Reply:
Ah - originally I did break it out but then we agreed not to do that after all https://github.com/NixOS/nixpkgs/pull/2482#issuecomment-42113475

Reply:
I modified the patching so that it’s more obvious which things are being patched on which platforms. For the time being I’m leaving things as they are with Handbrake’s dependencies: we use the Nixpkgs-provided versions, not the Handbrake patched versions.